pkg/chartutil/dependencies.go

```diff
@@ -19,15 +19,34 @@ import (
 	"log"
 	"strings"
 
+	"github.com/mitchellh/copystructure"
+
 	"helm.sh/helm/v3/pkg/chart"
 )
 
 // ProcessDependencies checks through this chart's dependencies, processing accordingly.
-func ProcessDependencies(c *chart.Chart, v Values) error {
-	if err := processDependencyEnabled(c, v, ""); err != nil {
+func ProcessDependencies(c *chart.Chart, vals map[string]interface{}) error {
```

Comment:
Given that this is a public exported function, we cannot change the function signature until Helm 4. Is there a way to rewrite this PR without requiring the need to rewrite the signature?

An alternative would be to introduce a new function and document `ProcessDependencies` as marked for deprecation in Helm 4.

Reply:
I've deprecated the ProcessDependencies method for ProcessDependenciesWithValues